evaluation/ragacli_pyproject/ragacli/pkg.py

```diff
@@ -0,0 +1,126 @@
+import os
+from datasets import Dataset
+from langchain.chat_models import ChatOpenAI
+from ragas.llms import RagasLLM
+from ragas.llms import LangchainLLM
+from ragas.embeddings import RagasEmbeddings
+from ragas.embeddings import OpenAIEmbeddings
+from ragas.embeddings import HuggingfaceEmbeddings
+from ragas.metrics.base import Metric
+
+from ragas.metrics import (
+    context_precision,
+    context_recall,
+    context_relevancy,
+    answer_relevancy,
+    answer_correctness,
+    answer_similarity,
+    faithfulness
+)
+
+DEFAULT_METRICS = [
+        "answer_relevancy",
+        "context_precision",
+        "faithfulness",
+        "context_recall",
+        "context_relevancy"
+    ]
+
+def wrap_langchain_llm(
+    model: str,
+    api_base: str | None,
+    api_key: str | None
+) -> LangchainLLM:
+    """
+    Initializes and returns an instance of the LangchainLLM class.
+
+    Args:
+        model (str): The name of the language model to use.
+        api_base (str | None): The base URL for the OpenAI API. If None, the default URL is assumed.
+        api_key (str | None): The API key for the OpenAI API.
+
+    Returns:
+        LangchainLLM: An instance of the LangchainLLM class.
+
+    Raises:
+        ValueError: If api_key is not provided.
+
+    Notes:
+        - If api_base is not provided, the default URL 'https://api.openai.com/v1' is assumed.
+        - The environment variables OPENAI_API_KEY and OPENAI_API_BASE are set to the provided api_key and api_base.
+    """
+    if api_base is None:
+        print('api_base not provided, assuming OpenAI default')
+        api_base = 'https://api.openai.com/v1'
+        os.environ["OPENAI_API_KEY"] = api_key
+        if api_key is None:
+            raise ValueError("api_key must be provided")
+        base = ChatOpenAI(model_name=model)
+    else:
+        os.environ["OPENAI_API_KEY"] = api_key
+        os.environ["OPENAI_API_BASE"] = api_base
+        base = ChatOpenAI(
+            model_name=model,
```

Comment:
since you already have this name configured in `ChatOpenAI`, can we remove `os.environ'

Reply:
ragas lib reaches out to `os.environ` as default, so this is rather like an 'insurance'? We can remove it tho